Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,284 @@
+        }
+        /// <summary>
+        /// The validate task that validate multiple input schemas with custom rules in JSchema.
+        /// </summary>
+        /// <param name="schemas">The schemas to be validated</param>
+        /// <param name="rules">The rules used to validate input schemas</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemas(Dictionary<string, ForgeTree> schemas, JSchema rules, out List<string> errorList)
+        {
+            return Validate(ConvertDictionaryToForgeTreeList(schemas), rules, out errorList);
+        }
+        /// <summary>
+        /// The validate task that validate the schema in the input file path with custom rules in JSchema.
+        /// </summary>
+        /// <param name="path">The path that contains a schema file</param>
+        /// <param name="rules">The rules used to validate input schemas</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaInPath(string path, JSchema rules, out List<string> errorList)
+        {
+            var schema = GetSchemaFromPath(path, out errorList);
+            return CheckConvertErrorAndValidate(rules, ref errorList, schema);
+        }
+        /// <summary>
+        /// The validate task that validate all schemas in a directory with custom rules in JSchema.
+        /// </summary>
+        /// <param name="path">The path that contains a schemas directory</param>
+        /// <param name="rules">The rules used to validate input schemas</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateMultipleSchemasInPath(string path, JSchema rules, out List<string> errorList)
+        {
+            var schemas = GetAllSchemasInDirectory(path, out errorList);
+            return CheckConvertErrorAndValidate(rules, ref errorList, schemas);
+        }
+
+        /// <summary>
+        /// The validate task that check the input schema in string with custom rules in JSchema.
+        /// </summary>
+        /// <param name="schema">The schema to be validated</param>
+        /// <param name="rules">The rules used to validate input schemas</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaString(string schema, JSchema rules, out List<string> errorList)
+        {
+            var schemaList = ConvertStringToJObjectList(schema, out errorList);
+            if (errorList.Count > 0)
+            {
+                return false;
+            }
+            var res = Validate(schemaList, rules, out errorList); 
+            return res;
```

Comment:
nit style:
1. Not a huge fan of var. It's okay in unittests, but I typically leave it out of code. Especially if the type isn't strongly typed in the same line. e.g. we don't explicitly know what type is returned from ConvertStringToJObjectList (even if it is obvious).

2. This one can vary from case to case, I don't have a clear ruleset for it. Sometimes it looks nice to add a newline above if/for/while statements to separate it from declarations.

3. Generally you want to have a newline after a closing brace. There are some exceptions, such as if/else statements or try/catch, or when followed by another closing brace. Besides that, if there is a line of code or a comment, add a newline.

4. Always have a newline above the return statement. Few exceptions here, for example, don't add a newline if the method is only one line.